### ISSUE TITLE
NTR:relationship for specifying stage of an individual

### ISSUE THREAD

**Issue by Clare72** [oborel/obo-relations#598]
Current 'existence overlaps' relationship is weird for an individual sample taken from an organism at a particular stage.
I think what we need is a 'has stage' relationship to capture the stage of life than an entity is currently in.

label: 'has stage'
definition: "A relationship between an anatomical entity and its current stage of life."
example of usage: "A tissue sample collected during the adult stage 'has stage' some 'adult stage'."

@dosumis

**Comment by nlharris** [comment]
@dosumis any comment on this?

**Comment by balhoff** [comment]
Discussed at RO meeting 2022-06-28: it might be useful to spell out the modeling for the example sentence (as some object property assertions). Would this relationship be derived from a property chain involving a collection event simultaneous with an adult stage? (for example)

**Comment by ddooley** [comment]
Currently to say an instance of an organism has a life cycle stage, we'd use "occurs in" right?

<img width="490" alt="image" src="https://user-images.githubusercontent.com/4000582/176259487-00af0a79-b7d4-48a1-a583-5dfbc39dbe55.png">

We could add "has stage" with domain organism and range life cycle stage?  That would jive with common language.

Then the problem that we're talking about is specimen deriving from some organism, so a shortcut relation. And that "life cycle stage" is not a (history) process but rather a spatiotemporal space that a process unfolds in.

For maximum clarity this suggests to me shortcut would be named "has source organism life stage".

**Comment by cmungall** [comment]
@Clare72 would this work:

`tissue and `existence starts and ends with` some S and derives_from some A`

(RO:0002491 "existence starts and ends with")

can you say more about how you would use the relation and whether a shortcut is required?

**Comment by ddooley** [comment]
@Clare72  would you happen to be available on RO call Tues Jan 31 to address this?  Issue is on agenda: http://bit.ly/ro-minutes

**Comment by Clare72** [comment]
> would you happen to be available on RO call Tues Jan 31 to address this?

ok

**Comment by ddooley** [comment]
processual entity -> "life cycle stage" doesn't follow aristotelean format 

Jie: let "life cycle stage" (UBERON) ..  be a process
useful to have specific relationship. 
"at life cycle stage" OR "at stage" OR "collected at life cycle stage" 

Damion will work on draft definition for:
[biosample][organism or part of one] "at life cycle stage" [life cycle stage]

**Comment by ddooley** [comment]
label: at life cycle stage
definition: A relationship between an anatomical entity and its current stage of life.
domain: cell, or gross anatomical part, or organism
range: life cycle stage
example: Embryo 'at life cycle stage' some 'embryo stage'
example: A tissue sample collected during the adult stage 'has stage' some 'adult stage'.
inverse: life cycle stage of

Question: Should COB adopt CARO "anatomical entity" to cover the disjunction of cell, gross anatomical part, or organism.  Defn could be tweaked: "Biological entity that is either an individual member of a biological species or constitutes part of the structural organization of an individual member of a biological species. "

**Comment by cmungall** [comment]
I'm pretty confused by this issue

At first we were talking about relating a sample to a stage at which the organism was at, in which case I agree with @ddooley that a shortcut relation chaining derives_from and (simultaneous_with or the relevant anatomical structure to stage relationships).

But if we are talking about relating the anatomical structure to the stage is this not covered in #611?

Additional points:

 - CARO will go away, existing generic classes in uberon are being generalized to be pan-life rather than pan-metazoa
 - for discussion of inclusion of stages in COB see https://github.com/OBOFoundry/COB/issues/40 (but not we don't need to solve this for RO purposes)

**Comment by ddooley** [comment]
For "**at life cycle stage**" definition, given its an output of specimen collection process, I think the label should be specialized to mention "specimen" too.

label: **specimen at life cycle stage**'

and definition too: `A relationship between an organism specimen and the organism's life cycle stage during which specimen collection occurred.`.

BTW, I can imagine biologists might like to say something specific about an instance of an organism in a life cycle stage but not imply anything necessarily about sample collection.  I think they can use the "has life cycle stage" at the instance level for this.

**Comment by wdduncan** [comment]
Do you think the label `specimen collected during life cycle stage` or (the shorter label) `collected during life cycle stage` would be better to emphasize the collection process?

**Comment by ddooley** [comment]
I like your 2nd label best!

**Comment by ddooley** [comment]
@wdduncan are you the giver-outer of RO ids?  I will mint some with a pull request. Also can I be a RO collaborator? (Otherwise it seems my obo-relations repo seems to be 1 step removed, and I don't quite know how to sync master over time.)

**Comment by wdduncan** [comment]
Hah! I had no idea that I had this role :)  
I was only helping out some external requests with ids.  
I'll create a PR modifying `ro-idranges.owl`. I'll partition my current id space. That is:
```
Datatype: idrange:19
Annotations: allocatedto: "Bill Duncan"
EquivalentTo: xsd:integer[> 17000 , <= 17500]

Datatype: idrange:25
Annotations: allocatedto: "Damion Dooley"
EquivalentTo: xsd:integer[> 17500 , <= 18000]
```

**Comment by ddooley** [comment]
So I see the id range pull request is ready to merge. Shall I?

I can't remember how to do a ro "make" locally on my computer to try my ro-edit.owl changes on?  I have to trigger new OBI "sample from organism", and UBERON "life cycle stage" as imports before I can add them to ranges. I was trying "    sh run.sh make release" per README-editors.md but that generates an error: "make: *** No rule to make target 'release'.  Stop."

**Comment by anitacaron** [comment]
@ddooley,

> I tried sh run.sh make update-uberon and that worked!

Do you mean `refresh-uberon`?

Yes, the docs need updates to add the ODK approach.

**Comment by ddooley** [comment]
Ah yes that was it, refresh-uberon.

**Comment by ddooley** [comment]
So further RO discussion gave the go-ahead to "**has life cycle stage**" and its inverse, which I have added as "**live cycle stage of**" in the pull request.

Discussion led to wanting to drop "**specimen at life cycle stage**" if there was another possible existing RO relation that covers this situation. 

I'm seeing an RO "**existence starts during**" which is a subclass of "**relation between physical entity and a process or stage**" defn. "... a diverse set of relations, typically connecting an anatomical entity to a biological process or developmental stage." So we could say "**specimen [x] "existence starts during" [life stage]**"?!  Is that fudging it?  Here we're talking about the existence of the specimen, rather than the morphological part the specimen derived from.  Seems like that semantic is consistent.

We can also add other statements such as "specimen [x] 'has taxon' some taxon [y]" and maybe helpfully add that "specimen [x] 'derives from' [anatomical part]".

**Comment by addiehl** [comment]
Chris M. suggests 'collected during'.  Damion will make a PR.

**Comment by ddooley** [comment]
Discussion revolved around wanting to shy away from reuse of "existence starts during" which is talking about in-place (in silico) development stage temporality, vs extracted material, and secondly generalizing the relationship to cover any processual entity, not just life cycle stages.  So "collected during" would cover this generally!

**Comment by ddooley** [comment]
Added "collected during" pull request https://github.com/oborel/obo-relations/pull/759 .  But what is its reverse? "Occured while collecting" ?

**Comment by wdduncan** [comment]
Perhaps add "developmental stage" to end of label -> "collected during developmental stage". 

Inverse would be: "developmental stage when collected"

Too wordy?

**Comment by ddooley** [comment]
We wanted to back off the specificity of having this relation attach to developmental stage.  Instead it would just pertain to any occurrent, so other circumstances during which the specimen was collected, e.g. high tide.  Ok?

**Comment by wdduncan** [comment]
> other circumstances during which the specimen was collected, e.g. high tide. Ok?

Ah yes ... Sorry ... brain cramps ... Now I seem to recall the conversation.

**Comment by wdduncan** [comment]
I apologize. But, I can't recall what the issue was with `happens during`. Can you remind me? It seems to work: `collection process happens during occurrent`.

**Comment by ddooley** [comment]
"Happens during" looks like its between two occurrents. "collected during" is between a material entity and an occurrent so it is a bit of a shortcut.

**Comment by wdduncan** [comment]
Another potential label for a more general relation between a material entity and occurrent: `existed during` or `existed during occurrent`.

**Comment by ddooley** [comment]
Hmm. "collection process" is a planned process, so "has specified output" is acceptable.  But I'm fine checking with the group about simplifying to "has input/output". At moment I vaguely recall issue is that "has specified input / output" can't be a sub-property of "has input/output" which then leads to conflict, but maybe that's getting resolved in COB?

If you propose changing label of "**happens during**" lets do that in another thread since that's an existing relationship.  Same with NTR "**existence occurs in**".

I was wondering if there was a more generic term for "collected". Collected does convey intentionality, and hints at a goal - a collection of things. "**Obtained**" perhaps, leaving vague about how obtained, and if destined for a collection.  "Selected" perhaps, but it doesn't imply possible extraction steps involved.  Acquired can imply money exchange.  Gathered almost implies gathering by hand; accrued implies accruing over time.

**Comment by wdduncan** [comment]
> Hmm. "collection process" is a planned process, so "has specified output" is acceptable. 

Not sure what the right path is here. Neither planned process or specified input relations are in RO. So, he relations seems really specific to OBI. We can add it to RO, but would not be able to define the domain, range, or property chain. 

We (in RO) need to get a better idea on what should be in RO and what should be left to other ontologies. E.g., Do we want all new relations defined in OBO ontologies to be in RO? I have mixed thoughts about this.

Question: Since this seems to be a very OBI specific relation, why do you think it should be in RO?

**Comment by ddooley** [comment]
Well, I am in favour of all relations being in RO for domain/range consistency, but I can see that could run into trouble as one gets deeper into sub-properties.

At any rate, the relation "collected during" doesn't have any dependencies directly on OBI classes.  A property chain would be nice but that can come later if/when RO does have all relations.

**Comment by wdduncan** [comment]
> At any rate, the relation "collected during" doesn't have any dependencies directly on OBI classes. A property chain would be nice but that can come later if/when RO does have all relations.

Yes, it can have domain:material entity and range:occurent. This would be very general and may not satisfy you in the sense that it won't be possible to verify if a collection process was involved. But, if that is fine with you, it is fine with me. I'll have to think about whether I want to submit an NTR for `existed during`.

**Comment by ddooley** [comment]
I remember now, that "collected during" range "occurent" allows us to say both the collection process, AND to reference other events related to the sample collection time, e.g. "collected during 'low tide'"

**Comment by wdduncan** [comment]
> e.g. "collected during 'low tide'"  

Lots of debate about the what things like 'low tide', 'adult stage', etc. are. Though not perfect, I classify such things as processes or subprocesses within a larger process. E.g., Low tide the subprocess of the larger tidal process during which the water level was at the low point. Again, not perfect, but stays away from having to also model temporal regions (and the like).

**Comment by cmungall** [comment]
From RO call:

- we summarize the "existence starts/ends" relations used in Uberon
- For the sample we can make a narrow synonym "has life cycle stage" on http://purl.obolibrary.org/obo/RO_0002491 "existence starts and ends during"
- we should make the existing relations subproperties of participation
- we overviewed the existing system of taxon constraints
